### DISCUSSION TITLE
[react] Add `bigint` to `ReactNode` type

### DISCUSSION THREAD

**Discussion by otomad** [DefinitelyTyped/DefinitelyTyped#69912] [Issues with a @types package] [answered]
It's useful to add `bigint` in `ReactNode` type.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1c20553fd9da69247ff3ffe1206be8222439615a/types/react/index.d.ts#L478-L489

Suggestion:

```typescript
 type ReactNode = 
     | ReactElement 
     | string 
     | number 
     | bigint
     | Iterable<ReactNode> 
     | ReactPortal 
     | boolean 
     | null 
     | undefined 
     | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES[ 
         keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES 
     ]; 
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
This is not supported in React 18 but only Canary. In Canary, we already have types for it:

```tsx
/// <reference types="react/canary" />
import type {ReactNode} from 'react'

const node: ReactNode = 5n
```

[Playground Link](https://www.typescriptlang.org/play/?target=99#code/PQgEB4CcFMDNpgOwMbVAFwJ4AdoGcBeAIhgENl1hlTFTJMjRgA+AKAEsBbbAe0nQw40AbwBK0cugByPACbQAvqFiQenUAHIyFDa1bIeiPAMRzoALlDjJM+aAKgArIiA)
